File: Array&String/Python/three_consecutive_odds.py
class Solution:
    def threeConsecutiveOdds(self, arr: list[int]) -> bool:
        n = len(arr)
        if n < 3: return False
        for i in range(n-2):
            if all(map(lambda x: x & 1, arr[i : i+3])): return True
        return False

File: Array&String/JS/three_consecutive_odds.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var threeConsecutiveOdds = function(arr) {
    let cnt = 0;
    for (x of arr) {
        cnt = x & 1? cnt+1 : 0;
        if (cnt == 3) return true;
    }
    return false;
};